.github/workflows/deploy-main.yaml

```diff
@@ -0,0 +1,44 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ main ]
+
+concurrency:
+  group: deployment
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20
+    - uses: pnpm/action-setup@v3
+      with:
+        version: 8
+    - uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-east-2
+        aws-access-key-id: ${{ secrets.AMAZON_ACCESS_KEY }}
+        aws-secret-access-key: ${{ secrets.AMAZON_SECRET_KEY }}
+    - run: pnpm ci
+    - run: IS_PROD_BUILD=true pnpm build
+
```

Comment:
We need a step here which actually uploads the built files to the `s3://www.surrealdb.com/docs` directory (before the workflow below syncs across the different regions).